File: dart/structural/decorator/man/bin/winter.dart
import 'man.dart';

class Winter implements Man {
  Man man;
  List<String> clothes;

  Winter(Man manValue) {
    man = manValue;
    clothes = [
      'down jacket',
      'long johns',
      'cotton shoes'
    ];
  }

  @override
  String attributes() {
    man.attributes();
    final result = 'WINTER: clothes = ${clothes[0]}, ${clothes[1]}, ${clothes[2]}';
    print(result);
    return result;
  }
}

File: typescript/structural/decorator/man/winter.ts
import {Man} from "./man";

export class Winter implements Man {
  man: Man
  clothes: string[]

  constructor(man: Man) {
    this.man = man
    this.clothes = [
      'down jacket',
      'long johns',
      'cotton shoes'
    ]
  }

  attributes(): string {
    console.log(this.man.attributes())
    return `WINTER MAN: clothes = ${this.clothes[0]}, ${this.clothes[1]}, ${this.clothes[2]}`
  }
}
